core/src/main/java/com/onelogin/saml2/http/HttpRequest.java

```diff
@@ -110,6 +144,38 @@ public String getParameter(String name) {
         return parameters;
     }
 
+    /**
+     * Return an url encoded get parameter value
+     * Prefer to extract the original encoded value directly from queryString since url
+     * encoding is not canonical.
+     * 
+     * @param name
+     * @return the first value for the parameter, or null
+     */
+    public String getEncodedParameter(String name) {    	
+    	Matcher matcher = Pattern.compile(Pattern.quote(name) + "=([^&]+)").matcher(queryString);
```

Comment:
nit: was gonna point out this should stop matching the parameter value at `#` to avoid including the fragment value, but probably all the querystrings passed here wouldn't have fragments.